wagtail/users/tests/test_admin_views.py

```diff
@@ -380,6 +380,25 @@ def test_create_with_custom_form(self):
         self.assertEqual(users.first().country, "testcountry")
         self.assertEqual(users.first().attachment.read(), b"Uploaded file")
 
+    def test_create_with_whitespaced_password(self):
+        """Password should not be stripped"""
+        self.post(
+            {
+                "username": "testuser2",
+                "email": "test@user2.com",
+                "first_name": "Test",
+                "last_name": "User",
+                "password1": "  whitespaced_password  ",
+                "password2": "  whitespaced_password  ",
+            },
+            follow=True,
+        )
+        # Try to login with the password
+        self.client.logout()
+        self.assertTrue(
+            self.client.login(username="testuser2", password="  whitespaced_password  ")
+        )
```

Comment:
The failing test uses the email-based custom user model: https://github.com/wagtail/wagtail/blob/29d73a3b689995f30ef998a33e079218340683a2/wagtail/test/emailuser/models.py#L35-L45

You can reuse the `login` method from our `WagtailTestUtils` class:

https://github.com/wagtail/wagtail/blob/29d73a3b689995f30ef998a33e079218340683a2/wagtail/test/utils/wagtail_tests.py#L78-L83

For example:

```suggestion
        self.login(username="testuser2", password="  whitespaced_password  ")
```